pkg/snapstore/snapstore_test.go

```diff
@@ -278,6 +280,220 @@ var _ = Describe("Save, List, Fetch, Delete from mock snapstore", func() {
+	},
+	// Azure
+	{
+		Provider:          "ABS",
+		EnvVariable:       "AZURE_APPLICATION_CREDENTIALS",
+		SnapstoreProvider: brtypes.SnapstoreProviderABS,
+		CredentialType:    "directory",
+		CredentialFiles:   []string{"storageAccount", "storageKey"},
+	},
+	{
+		Provider:          "ABS",
+		EnvVariable:       "AZURE_APPLICATION_CREDENTIALS_JSON",
+		SnapstoreProvider: brtypes.SnapstoreProviderABS,
+		CredentialType:    "file",
+		CredentialFiles:   []string{"credentials.json"},
+	},
+	// GCS
+	{
+		Provider:          "GCS",
+		EnvVariable:       "GOOGLE_APPLICATION_CREDENTIALS",
+		SnapstoreProvider: brtypes.SnapstoreProviderGCS,
+		CredentialType:    "file",
+		CredentialFiles:   []string{"credentials.json"},
+	},
+	// Swift V3ApplicationCredentials
+	{
+		Provider:          "Swift",
+		EnvVariable:       "OPENSTACK_APPLICATION_CREDENTIALS",
+		SnapstoreProvider: brtypes.SnapstoreProviderSwift,
+		CredentialType:    "directory",
+		CredentialFiles:   []string{"authURL", "tenantName", "domainName", "applicationCredentialID", "applicationCredentialName", "applicationCredentialSecret"},
+	},
+	// Swift Password
+	{
+		Provider:          "Swift",
+		EnvVariable:       "OPENSTACK_APPLICATION_CREDENTIALS",
+		SnapstoreProvider: brtypes.SnapstoreProviderSwift,
+		CredentialType:    "directory",
+		CredentialFiles:   []string{"authURL", "tenantName", "domainName", "username", "password"},
+	},
+	// Swift JSON
+	{
+		Provider:          "Swift",
+		EnvVariable:       "OPENSTACK_APPLICATION_CREDENTIALS_JSON",
+		SnapstoreProvider: brtypes.SnapstoreProviderSwift,
+		CredentialType:    "file",
+		CredentialFiles:   []string{"credentials.json"},
+	},
+	// OSS
+	{
+		Provider:          "OSS",
+		EnvVariable:       "ALICLOUD_APPLICATION_CREDENTIALS",
+		SnapstoreProvider: brtypes.SnapstoreProviderOSS,
+		CredentialType:    "directory",
+		CredentialFiles:   []string{"accessKeyID", "accessKeySecret", "storageEndpoint"},
+	},
+	{
+		Provider:          "OSS",
+		EnvVariable:       "ALICLOUD_APPLICATION_CREDENTIALS_JSON",
+		SnapstoreProvider: brtypes.SnapstoreProviderOSS,
+		CredentialType:    "file",
+		CredentialFiles:   []string{"credentials.json"},
+	},
+	// OCS
+	{
+		Provider:          "OCS",
+		EnvVariable:       "OPENSHIFT_APPLICATION_CREDENTIALS",
+		SnapstoreProvider: brtypes.SnapstoreProviderOCS,
+		CredentialType:    "directory",
+		CredentialFiles:   []string{"accessKeyID", "region", "endpoint", "secretAccessKey"},
+	},
+	{
+		Provider:          "OCS",
+		EnvVariable:       "OPENSHIFT_APPLICATION_CREDENTIALS_JSON",
+		SnapstoreProvider: brtypes.SnapstoreProviderOCS,
+		CredentialType:    "file",
+		CredentialFiles:   []string{"credentials.json"},
+	},
+}
+
+var _ = Describe("Dynamic access credential rotation test for each provider", func() {
+	for _, config := range credentialTestConfigs {
+		config := config
+		Context(fmt.Sprintf("testing secret modification for %s with %s", config.Provider, config.EnvVariable), func() {
```

Comment:
```suggestion
		Describe(fmt.Sprintf("testing secret modification for provider %q with %q", config.Provider, config.EnvVariable), func() {
```

Reply:
Addressed in [8635c10](https://github.com/gardener/etcd-backup-restore/pull/670/commits/8635c101d031f890343da3177cb410aa20053dc5)